task_executor/scripts/scheduled_task_executor.py

```diff
@@ -259,11 +262,12 @@ def call_scheduler(self, tasks, earliest_start, current_id=0):
         Calls scheduler. Reorders the list of tasks in execution order with their execution times set. 
         
         """
-        resp = self.schedule_srv(tasks, earliest_start, current_id, self.get_duration_matrix(tasks))
+        resp = self.schedule_srv(tasks, earliest_start, current_id, self.get_duration_matrix(tasks)) # Lenka note: why does the service need earliest start and current id?
```

Comment:
To answer your note, because (I think) it's assuming that execution of the rest of the schedule must start after the current task finishes.